### PR TITLE
Create small patch domain on a sphere

### PR DESCRIPTION
**Pull request by mark-petersen**
In COMPASS, create a domain on the sphere that is simply all the cells within a lat/lon square.

### CODE REVIEW

File: testing_and_setup/compass/ocean/global_ocean/patch/init/cull_mesh_patch.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+landMask = numpy.logical_and(numpy.logical_and(lat >= minlat, lat <= maxlat),
+                             numpy.logical_and(lon >= minlon, lon <= maxlon))
+
+# Convert logical to int
+landMask = landMask.astype(int)
+
+dsLandMask = xarray.Dataset()
+dsLandMask['cullCell'] = landMask
+
+dsCulledMesh = conversion.cull(dsBaseMesh, dsMask=dsLandMask,
```
Comment (mark-petersen) [inline testing_and_setup/compass/ocean/global_ocean/patch/init/cull_mesh_patch.py]: @xylar, when I run this case: `-o ocean -c global_ocean -r patch -t init` it runs through successfully and creates `culled_mesh.nc`, but no cells are culled. Perhaps you can see why. Is this dsMask argument looking for an array called `cullCell` within the Dataset called dsLandMask? I'm not very familiar with xarray.

File: testing_and_setup/compass/ocean/global_ocean/patch/init/cull_mesh_patch.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+oceanMask = numpy.logical_and(numpy.logical_and(lat >= minlat, lat <= maxlat),
+                              numpy.logical_and(lon >= minlon, lon <= maxlon))
+
+# Convert logical to int
+landMask = numpy.logical_not(oceanMask).astype(int)
+
+dsBaseMesh['cullCell'] = landMask
+
+dsCulledMesh = conversion.cull(dsBaseMesh,
+                               graphInfoFileName='culled_graph.info')
```
Comment (xylar) [inline testing_and_setup/compass/ocean/global_ocean/patch/init/cull_mesh_patch.py]: @mark-petersen, `cullCell` is added to the base mesh, not passed as a separate mask dataset.  (If you want to do the latter, you need to call the field `regionMasks` and it needs to have an extra dimension that we don't want to bother with in this case.)

Also, `cullCells` is 1 where we get rid of cells and 0 where we keep them, so I made some changes accordingly.

**Comment by xylar** [comment]
@mark-petersen, I made some changes so cell culling works properly.  Happy to explain more if needed when you get back.  I ran the full test case and it didn't crash...  I didn't look at the output beyond the `cull_mesh` stage so I'll leave it to you to double check before merging.  Let me know if you need more testing from me.

**Comment by xylar** [comment]
@mark-petersen, please close this and optionally move it to https://github.com/MPAS-Dev/compass

**Comment by mark-petersen** [comment]
Closing. This PR was just something we tried, and is a low priority.

**Pull request closed without merging**